lib/ui/screens/group_screen.dart

```diff
@@ -0,0 +1,307 @@
+      ),
+    );
+  }
+}
+
+class _GroupImage extends StatelessWidget {
+  const _GroupImage({
+    Key? key,
+    required this.group,
+  }) : super(key: key);
+
+  final ListGroup group;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      fit: StackFit.loose,
+      children: [
+        CachedImage(
+          imageUrl: group.photo.full,
+          placeholder: 'assets/img/default-avatar.jpg',
+        )
+      ],
+    );
+  }
+}
+
+class _MembersGrid extends StatelessWidget {
+  const _MembersGrid({
+    Key? key,
+    this.members,
+  }) : super(key: key);
+
+  final List<GroupMembership>? members;
+
+  @override
+  Widget build(BuildContext context) {
+    if (members == null) {
+      return const SliverPadding(
+        padding: EdgeInsets.only(left: 16, right: 16, top: 8, bottom: 16),
+        sliver: SliverToBoxAdapter(
+          child: Center(child: CircularProgressIndicator()),
+        ),
+      );
+    } else if (members!.isEmpty) {
+      return const SliverPadding(
+        padding: EdgeInsets.only(left: 16, right: 16, top: 8, bottom: 16),
+        sliver: SliverToBoxAdapter(
+          child: Center(child: Text('This group has no members.')),
+        ),
+      );
+    } else {
+      return SliverPadding(
+        padding: const EdgeInsets.only(left: 16, right: 16, top: 8, bottom: 16),
+        sliver: SliverGrid(
+          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+            crossAxisCount: 3,
+            mainAxisSpacing: 8,
+            crossAxisSpacing: 8,
+          ),
+          delegate: SliverChildBuilderDelegate(
+            (context, index) {
+              return MemberTile(
+                member: members![index].member,
+              );
+            },
+            childCount: members!.length,
+          ),
+        ),
+      );
+    }
+  }
+}
+
+class _GroupInfo extends StatelessWidget {
+  const _GroupInfo({
+    Key? key,
+    required this.group,
+  }) : super(key: key);
+
+  final ListGroup group;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 4),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          const SizedBox(height: 8),
+          Text(
+            group.name.toUpperCase(),
+            style: Theme.of(context).textTheme.headline6,
+          ),
+          const Divider(height: 24),
+          _Description(group: group),
+          const Divider(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class _Description extends StatelessWidget {
+  const _Description({
+    Key? key,
+    required this.group,
+  }) : super(key: key);
+
+  final ListGroup group;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 0, vertical: 4),
+      child: HtmlWidget(
+        group.description,
+        onTapUrl: (String url) async {
```

Comment:
outline to a function

Reply:
I don't understand what you mean, either way I am pretty sure this is done the same way in other screens

Reply:
Yes it is done in other screens but having big functions inside a widget tree makes the tree hard to read.

The function should be a method, and then pass the method on to `onTapUrl`

Reply:
It would be nnice to outline this everywhere (it's the same everywhere, so moving it to a `routes.dart` or some other utility place would be DRY and easy to read). But that's a refactor that has nothing to do with groups, so best to do that later in a separate PR

Reply:
I'll track this in #99.